### DISCUSSION TITLE
Stanford CS110: Principles of Computer Systems - CS自学指南

### DISCUSSION THREAD

**Discussion by giscus** [PKUFlyingPig/cs-self-learning#598] [Announcements]
# Stanford CS110: Principles of Computer Systems - CS自学指南

CS自学指南

https://csdiy.wiki/%E8%AE%A1%E7%AE%97%E6%9C%BA%E7%B3%BB%E7%BB%9F%E5%9F%BA%E7%A1%80/CS110/

**Comment by Owenun** [comment]
这门课的lab和assignment的代码是开放的吗?

**Comment by AdingApkgg** [comment]
test

**Comment by JasonZhaoWW** [comment]
学习这门课是不是需要先学CS107

**Comment by Lanisera** [comment]
有人知道assign1中的sanitycheck在哪里吗，是可以用slink里的三个sanity文件生成吗

**Comment by Keedled** [comment]
上新了，支持

**Comment by Andy-xiaokang** [comment]
这门课是什么都做不了吗？

**Comment by ghost** [comment]
cs107有没有提供课程啊

**Comment by ArvidJia** [comment]
这个仓库里面有Assignment与lab的资源：https://github.com/sauronalexander/CS110-2

**Comment by tommy0103** [comment]
CS110 的确非常好，但是关于框架代码并不是很好获取，最后一个 assignment8 的代码找了很久，而且也很难找到 starter code
https://github.com/xiaoyubai/cs110/tree/master
比较推荐这个 repository，对于 assign 有单独的 docker 环境，并且有 assign8 的 start code
从 assign2 写到 assign7，感觉收获很大，印象最深刻的是 ThreadPool 的实现那个 assign 和 HTTPProxy，ThreadPool 的实现中会遇到很多 Race Condition，非常难处理，而无论是多线程的 HTTPProxy 还是线程池的 Proxy 实现，过程中都会因为科学上网等等问题而十分困扰，以及框架代码中的 xmlReadFile 十分欠缺稳定性，可能需要重写 libcurl + xmlReadMemory 的代码 
但是比较不好的一点是，assign8 的 MapReduce 和 assign7 的 proxyChain 都是基于 myth 分布式环境搭建的框架代码，如果是个人学习没有环境会很难检验代码正误

**Reply by 1740907592** [reply]
docker运行好像会出错

**Reply by tommy0103** [reply]
是这样的，我没使用 docker 环境

**Reply by tommy0103** [reply]
我猜是 Winter 2022，但我看的视频似乎并不是这个版本
各个版本之间讲的内容都大差不差，无非是 assign 的内容发生了变化

**Comment by YixinXue** [comment]
这个课教的是真的好，比csapp细致多了
